app/src/main/java/at/bitfire/icsdroid/model/CredentialsModel.kt

```diff
@@ -0,0 +1,22 @@
+package at.bitfire.icsdroid.model
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class CredentialsModel : ViewModel() {
+    var originalRequiresAuth: Boolean? = null
+    var originalUsername: String? = null
+    var originalPassword: String? = null
+
+    val requiresAuth = MutableLiveData<Boolean>()
+    val username = MutableLiveData<String>()
+    val password = MutableLiveData<String>()
+
+    init {
+        requiresAuth.value = false
+    }
+
+    fun dirty() = requiresAuth.value != originalRequiresAuth ||
```

Comment:
At some point I think we should really get rid of the `dirty` function as use a `MediatorLiveData` to check the states.

When migrating to Compose I believe it will be more "reactive".

Reply:
Agree! I started on it in [the PR for the `SubscriptionSettingsFragment`](https://github.com/ArnyminerZ/icsx5/pull/29/files#diff-4d77a4cfab3683b0123ba975d0c00894bfc88c0ffaadd3280faca6df2e6ff7afR223-R224)

Reply:
Guess, I should do it here too :)

Reply:
We can wait for #29, there's no hurry right now, but just to have it in mind for the future